modules/gapi/src/compiler/gislandmodel.cpp

```diff
@@ -335,6 +335,25 @@ ade::NodeHandle GIslandModel::producerOf(const ConstGraph &g, ade::NodeHandle &d
     return ade::NodeHandle();
 }
 
+std::string GIslandModel::traceIslandName(const ade::NodeHandle& island_nh, const Graph& g) {
+    auto island_ptr = g.metadata(island_nh).get<FusedIsland>().object;
+    std::string island_name = island_ptr->name();
+
+    std::string backend_name = "";
+    auto& backend_impl = island_ptr->backend().priv();
+    std::string backend_impl_type_name = typeid(backend_impl).name();
+    std::regex rgx("G([a-zA-Z0-9]+)BackendImpl");
```

Comment:
`std::regex` is not working in GCC<4.9 (https://stackoverflow.com/questions/12530406/is-gcc-4-8-or-earlier-buggy-about-regular-expressions), but we still kinda support CentOS 7 which uses GCC 4.8.5. Is it possible to rewrite this part without regex usage?